cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/batch/mapreduce/MapReduceBatchContext.java

```diff
@@ -100,6 +100,38 @@ public T call() throws Exception {
   }
 
   @Override
+  public void releaseDataset(final Dataset dataset) {
+    try {
+      LogContext.runWithoutLogging(new Callable<Void>() {
+        @Override
+        public Void call() {
+          mrContext.releaseDataset(dataset);
+          return null;
+        }
+      });
+    } catch (Exception e) {
+      Throwables.propagateIfInstanceOf(e, DatasetInstantiationException.class);
+      throw Throwables.propagate(e);
+    }
+  }
+
+  @Override
+  public void discardDataset(final Dataset dataset) {
+    try {
+      LogContext.runWithoutLogging(new Callable<Void>() {
+        @Override
+        public Void call() {
+          mrContext.discardDataset(dataset);
+          return null;
+        }
+      });
+    } catch (Exception e) {
```

Comment:
There's the LogContext#runWithoutLoggingUnchecked which is meant to be used when you need to propagate exceptions like this, right?